File: sdks/dotnet/sulfone-helium/Api/Plugin/Mapper.cs
using sulfone_helium.Domain.Plugin;

namespace sulfone_helium.Api.Plugin;

public static class PluginMapper
{
    public static PluginInput ToDomain(this PluginReq req)
    {
        return new PluginInput(req.Directory, req.Config.ToDynamic());
    }

    public static PluginRes ToRes(this PluginOutput res)
    {
        return new PluginRes { OutputDir = res.Directory };
    }
}

File: sdks/python/cyan_sdk/api/plugin/mapper.py
from cyan_sdk.api.plugin.req import PluginReq
from cyan_sdk.api.plugin.res import PluginRes
from cyan_sdk.domain.plugin.input import PluginInput
from cyan_sdk.domain.plugin.output import PluginOutput


class PluginMapper:
    @staticmethod
    def to_domain(req: PluginReq) -> PluginInput:
        return PluginInput(directory=req.directory, config=req.config)

    @staticmethod
    def to_res(res: PluginOutput) -> PluginRes:
        return PluginRes(output_dir=res.directory)
